subgraphs/eigenda-payments/src/payment-vault.ts

```diff
@@ -0,0 +1,170 @@
+  entity.totalDeposit = event.params.totalDeposit
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handleOwnershipTransferred(
+  event: OwnershipTransferredEvent
+): void {
+  let entity = new OwnershipTransferred(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity.previousOwner = event.params.previousOwner
+  entity.newOwner = event.params.newOwner
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handlePriceParamsUpdated(event: PriceParamsUpdatedEvent): void {
+  let entity = new PriceParamsUpdated(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity.previousMinNumSymbols = event.params.previousMinNumSymbols
+  entity.newMinNumSymbols = event.params.newMinNumSymbols
+  entity.previousPricePerSymbol = event.params.previousPricePerSymbol
+  entity.newPricePerSymbol = event.params.newPricePerSymbol
+  entity.previousPriceUpdateCooldown = event.params.previousPriceUpdateCooldown
+  entity.newPriceUpdateCooldown = event.params.newPriceUpdateCooldown
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handleReservationPeriodIntervalUpdated(
+  event: ReservationPeriodIntervalUpdatedEvent
+): void {
+  let entity = new ReservationPeriodIntervalUpdated(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity.previousValue = event.params.previousValue
+  entity.newValue = event.params.newValue
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+}
+
+export function handleReservationUpdated(event: ReservationUpdatedEvent): void {
+  let entity = new ReservationUpdated(
+    event.transaction.hash.concatI32(event.logIndex.toI32())
+  )
+  entity.account = event.params.account
+  entity.reservation_symbolsPerSecond =
+    event.params.reservation.symbolsPerSecond
+  entity.reservation_startTimestamp = event.params.reservation.startTimestamp
+  entity.reservation_endTimestamp = event.params.reservation.endTimestamp
+  entity.reservation_quorumNumbers = event.params.reservation.quorumNumbers
+  entity.reservation_quorumSplits = event.params.reservation.quorumSplits
+
+  entity.blockNumber = event.block.number
+  entity.blockTimestamp = event.block.timestamp
+  entity.transactionHash = event.transaction.hash
+
+  entity.save()
+
+  // Create or update the ActiveReservation entity for this account
+  let activeReservation = CurrentReservation.load(event.params.account)
+  if (activeReservation == null) {
+    activeReservation = new CurrentReservation(event.params.account)
```

Comment:
do these currentReservations ever get deleted? How does the subgraph know that its still current? DO we have to filter manually for endTimestamp?

Reply:
reservations never get deleted on-chain, so we need to filter. I would recommend adding a `status` enum type for `active`, `expired` or `inactive` since there could be reservation that gets activated in the future

Reply:
Will use the following logic
```
enum ReservationStatus {
  PENDING    # startTimestamp > current time
  ACTIVE     # startTimestamp <= current time < endTimestamp
  EXPIRED    # endTimestamp <= current time
}
```

Reply:
makes sense, that's okay with me

Reply:
Yep that makes a ton of sense, nice md! Can we just change `currentReservations` to simply `reservations` everywhere then, since expired reservations are not deleted, so they aren't "current"? Or maybe `registeredReservations` or something?